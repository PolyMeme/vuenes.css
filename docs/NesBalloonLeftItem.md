# nes-balloon-left-item 

Vue Component implementation of the Nes Balloon in the Left Side 

- **author** - Jesus Fernando Alvarez Franco 
- **license** - MIT 

## props 

- `message` ***String*** (*optional*) `default: Empty String` 

  Define the message of the balloon message 

