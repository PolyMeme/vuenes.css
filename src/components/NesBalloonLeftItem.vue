<template>
  <div class="message -left">
    <div class="nes-balloon from-left">
      <p>{{ message }}</p>
    </div>
  </div>
</template>

<script>
/**
 * Vue Component implementation of the Nes Balloon in the Left Side
 * @author Jesus Fernando Alvarez Franco
 * @license MIT
 */
export default {
  props: {
    /**
     * Define the message of the balloon message
     * @default Empty String
     */
    message: {
      type: String,
      default: () => ''
    }
  }
}
</script>
