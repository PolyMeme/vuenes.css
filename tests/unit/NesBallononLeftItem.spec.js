import { mount } from '@vue/test-utils'
import NesBalloonLeftItem from '../../src/components/NesBalloonLeftItem.vue'

describe('NesBalloonLeftItem.vue', () => {
  it('should render empty message', () => {
    const wrapper = mount(NesBalloonLeftItem)
    expect(wrapper.find('div.message.-left>div.nes-balloon.from-left>p').text()).toBe('')
  })

  it('should match snapshot', () => {
    const wrapper = mount(NesBalloonLeftItem, {
      propsData: {
        message: 'This is a message from the left!'
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
})
