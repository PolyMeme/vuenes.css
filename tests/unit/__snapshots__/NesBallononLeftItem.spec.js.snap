// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NesBalloonLeftItem.vue should match snapshot 1`] = `
<div class="message -left">
  <div class="nes-balloon from-left">
    <p>This is a message from the left!</p>
  </div>
</div>
`;
